front/app/containers/ProjectsShowPage/shared/header/ProjectActionButtons.tsx

```diff
@@ -51,193 +55,163 @@ interface Props {
 const ProjectActionButtons = memo<Props>(({ projectId, className }) => {
   const project = useProject({ projectId });
   const phases = usePhases(projectId);
-
   const [currentPhase, setCurrentPhase] = useState<IPhaseData | null>(null);
-  const [ideasPresentOutsideViewport, setIdeasPresentOutsideViewport] =
-    useState(false);
-  const [surveyPresentOutsideViewport, setSurveyPresentOutsideViewport] =
-    useState(false);
-  const [pollPresentOutsideViewport, setPollPresentOutsideViewport] =
-    useState(false);
+  const divId = clHistory.location.hash;
+  const timeIdRef = useRef<NodeJS.Timeout>();
 
   useEffect(() => {
     setCurrentPhase(getCurrentPhase(phases) || getLastPhase(phases));
   }, [phases]);
 
   useEffect(() => {
-    const loop = (counter: number) => {
-      if (counter < 200) {
-        setTimeout(() => {
-          const viewportHeight = Math.max(
-            document.documentElement.clientHeight || 0,
-            window.innerHeight || 0
-          );
-
-          const ideasElement = document.getElementById('project-ideas');
-          const surveyElement = document.getElementById('project-survey');
-          const pollElement = document.getElementById('project-poll');
-
-          if (ideasElement) {
-            const isIdeasInViewport =
-              ideasElement.getBoundingClientRect()?.top + 800 <= viewportHeight;
-
-            if (!isIdeasInViewport) {
-              setIdeasPresentOutsideViewport(true);
-            }
-          }
-
-          if (surveyElement) {
-            const isSurveyInViewport =
-              surveyElement.getBoundingClientRect()?.top + 400 <=
-              viewportHeight;
-
-            if (!isSurveyInViewport) {
-              setSurveyPresentOutsideViewport(true);
-            }
-          }
-
-          if (pollElement) {
-            const isPollInViewport =
-              pollElement.getBoundingClientRect()?.top + 200 <= viewportHeight;
-
-            if (!isPollInViewport) {
-              setPollPresentOutsideViewport(true);
-            }
-          }
-
-          loop(counter + 1);
-        }, 10);
-      }
-    };
+    const element = document.getElementById(divId);
+    if (element) {
+      element.scrollIntoView();
+    }
+  }, [divId]);
+
+  const clearSetTimeout = () => {
+    if (timeIdRef.current) {
+      clearTimeout(timeIdRef.current);
+    }
+  };
 
-    loop(0);
-  }, [projectId]);
+  useEffect(() => {
+    return () => clearSetTimeout();
+  }, []);
 
   const scrollTo = useCallback(
     (id: string, shouldSelectCurrentPhase = true) =>
       (event: FormEvent) => {
         event.preventDefault();
 
-        currentPhase && shouldSelectCurrentPhase && selectPhase(currentPhase);
+        if (!isNilOrError(project)) {
+          const isOnProjectPage = window.location.pathname.endsWith(
```

Comment:
Please use react router for these types of checks :) `useLocation` is the hook for this one.